lib/swing-lib/src/main/java/org/triplea/ui/events/queue/ViewControllerEventQueue.java

```diff
@@ -0,0 +1,53 @@
+package org.triplea.ui.events.queue;
+
+import com.google.common.base.Preconditions;
+import java.util.function.UnaryOperator;
+import lombok.Setter;
+
+/**
+ * Adapted for Swing!
+ *
+ * <p>Queue to pass events between a Swing UI and a view-model (AKA presentation layer).
+ *
+ * <p>UIs publish UI events and Model publish model events. UI listen for model events and models
+ * listen for UI events. The model event callbacks (UI) are executed on the Swing EDT thread, and UI
+ * event callbacks are executed in a new thread (off of the EDT).
+ */
+@Setter
+public class ViewControllerEventQueue<
```

Comment:
Maybe I misunderstood the purpose of this class, but I expected to find `publishUiEvent` invoking the callback on a new thread and `publishModelEvent` invoking the callback on the EDT; as it stands, they both invoke their respective callbacks on the current thread. Is this simply an oversight, or is the thread affinity code someplace else and I just missed it?

Reply:
You're right, and following along nicely   = D
The thread affinity code was built in a first revision of this class and this class is a rewrite and simply hasn't gotten back to having the affinity code. It turns out it is not that difficult, and then blissful to know in the handlers that everything is on the right thread.

I'll comment and point out when that logic is added back.